src/Plugins/ITKImageProcessing/src/ITKImageProcessing/Filters/ITKMhaFileReader.cpp

```diff
@@ -0,0 +1,441 @@
+#include "ITKMhaFileReader.hpp"
+
+#include <filesystem>
+
+#include "itkMetaImageIO.h"
+
+#include "ITKImageProcessing/Filters/ITKImageReader.hpp"
+
+#include "complex/Core/Application.hpp"
+#include "complex/DataStructure/Geometry/ImageGeom.hpp"
+#include "complex/Filter/Actions/CreateArrayAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/AttributeMatrixSelectionParameter.hpp"
+#include "complex/Parameters/BoolParameter.hpp"
+#include "complex/Parameters/ChoicesParameter.hpp"
+#include "complex/Parameters/DataGroupCreationParameter.hpp"
+#include "complex/Parameters/DataObjectNameParameter.hpp"
+#include "complex/Parameters/DynamicTableParameter.hpp"
+#include "complex/Parameters/FileSystemPathParameter.hpp"
+#include "complex/Parameters/GeometrySelectionParameter.hpp"
+
+namespace fs = std::filesystem;
+
+using namespace complex;
+
+namespace
+{
+const Uuid k_ComplexCorePluginId = *Uuid::FromString("05cc618b-781f-4ac0-b9ac-43f26ce1854f");
+const Uuid k_ApplyTransformationToGeometryFilterId = *Uuid::FromString("f5bbc16b-3426-4ae0-b27b-ba7862dc40fe");
+const FilterHandle k_ApplyTransformationToGeometryFilterHandle(k_ApplyTransformationToGeometryFilterId, k_ComplexCorePluginId);
+
+static inline constexpr StringLiteral k_SelectedImageGeometry_Key = "selected_image_geometry";
+static inline constexpr StringLiteral k_CellAttributeMatrixPath_Key = "cell_attribute_matrix_path";
+static inline constexpr StringLiteral k_TransformationType_Key = "transformation_type";
+static inline constexpr StringLiteral k_ManualTransformationMatrix_Key = "manual_transformation_matrix";
+static inline constexpr StringLiteral k_InterpolationType_Key = "interpolation_type";
+const complex::ChoicesParameter::ValueType k_ManualTransformationMatrixIdx = 2ULL;
+
+const std::string k_NearestNeighborInterpolation("Nearest Neighbor");
+const std::string k_LinearInterpolation("Linear Interpolation");
+const complex::ChoicesParameter::Choices k_InterpolationChoices = {k_NearestNeighborInterpolation, k_LinearInterpolation};
+
+const complex::ChoicesParameter::ValueType k_NearestNeighborInterpolationIdx = 0ULL;
+const complex::ChoicesParameter::ValueType k_LinearInterpolationIdx = 1ULL;
+
+Result<std::array<float32, 16>> ReadTransformMatrix(const std::string& filePath)
+{
+  std::array<float32, 16> arr;
```

Comment:
this should be initialized to something .. maybe all zeros?

Reply:
That is for global (Static) storage. Your use is on the stack... Look at the next example on that web page